Shared/Coordinator/Coordinator.swift

```diff
@@ -8,6 +8,7 @@
 
 import Foundation
 
+@MainActor
```

Comment:
All coordinators are now run on the MainActor. This is because Coordinators are primarily running UI code which must are run on the main thread.